### DISCUSSION TITLE
default value of each clock source with STM32H750

### DISCUSSION THREAD

**Discussion by nopnop2002** [stm32duino/Arduino_Core_STM32#2220] [Q&A]
The STM32H750 FDCAN clock source can be selected from the following.
```
RCC domain 2 kernel clock configuration register
(RCC_D2CCIP1R) 

Bits 29:28 FDCANSEL: FDCAN kernel clock source selection
Set and reset by software.
00: hse_ck clock is selected as FDCAN kernel clock (default after reset)
01: pll1_q_ck clock is selected as FDCAN kernel clock
10: pll2_q_ck clock is selected as FDCAN kernel clock
```

What is the default value of each clock source in Hz?

**Comment by fpistm** [comment]
Hi @nopnop2002 
it depends of the board. You can check with CubeMX and the related system core clock config

**Comment by nopnop2002** [comment]
Is there a way to check using Arduino core support?

**Reply by fpistm** [reply]
You can probably use the HAL or LL RCC to get the value you want.
Ex:
https://github.com/stm32duino/Arduino_Core_STM32/blob/b149a92d074cbf3ac557317a665a0d971996691a/system/Drivers/STM32H7xx_HAL_Driver/Inc/stm32h7xx_hal_rcc_ex.h#L3276C32-L3286

**Reply by nopnop2002** [reply]
I checked the clock source value of FDCAN with the sketch below on STM32H750VBTx.

As a result, the following was displayed:

F_CPU=480000000
fdcan_clock=536870912

Is this correct?


```
void setup() {
  delay(1000);
  Serial.begin(115200);
  RCC->D2CCIP1R &= ~(0x30000000UL);    // clear FDCAN kernel clock source selection
  RCC->D2CCIP1R |= 0x20000000UL;       // PLL2_q_ck clock is selected as FDCAN kernel clock
  long fdcan_clock = __HAL_RCC_GET_FDCAN_SOURCE();
  Serial.print("F_CPU=");
  Serial.println(F_CPU);
  Serial.print("fdcan_clock=");
  Serial.println(fdcan_clock);
}

void loop() {
  // put your main code here, to run repeatedly:

}
```

**Reply by fpistm** [reply]
well, don't really understand why you ask this as 536870912 is equal to 0x20000000UL and it is what you set. Which is `RCC_FDCANCLKSOURCE_PLL2`.

**Reply by nopnop2002** [reply]
I would like to know the actual clock frequency [Hz] of PLL2_q_ck like F_CPU macro.
